### PR TITLE
Support User: Refactor to remove singleton class, use individual fn exports instead (fixed)

### PR DESCRIPTION
**Pull request by jordwest**
This change fixes the issue in #3587 that broke staging.

Commit d4118c0 contains the original PR content, while the changes in 066e263 solve the issue that broke staging.

To confirm that this fixes the issue, test with both:
```
make run
```
and
```
DISABLE_FEATURES=support-user make run
```

cc @dllh @gwwar

### CODE REVIEW

File: client/lib/user/support-user-interop.js [hunk 1]
```diff
@@ -9,11 +9,9 @@ import debugModule from 'debug';
 import wpcom from 'lib/wp';
 import config from 'config';
 import store from 'store';
+import noop from 'lodash/noop';
```
Comment (gwwar) [inline client/lib/user/support-user-interop.js]: The store and lodash import should be grouped with the external dependencies.

File: client/boot/index.js [hunk 2]
```diff
@@ -1,4 +1,9 @@
 /**
+ * Internal dependencies
+ */
+import { setReduxStore as setSupportUserReduxStore } from 'lib/user/support-user-interop';
```
Comment (gwwar) [inline client/boot/index.js]: Maybe leave this grouped with the existing internal dependencies, and we can add a separate janitorial PR to update import usages.

**Comment by gwwar** [comment]
:+1: Sorry about the slip last time. I tested with both pathways this time, and verified that there were no functional changes.

If we're afraid of breakages in the future we can test both paths by stubbing the config value for tests:
```javascript
const configStub = sinon.stub( config, 'isEnabled' );
configStub.withArgs( 'support-user' ).returns( false );
```

Otherwise, this should be good to :ship: after a tiny bit of import tidying.

**Comment by jordwest** [comment]
@gwwar totally my mistake, I should have tested that before putting it up for review.

I've fixed up those imports and tested again with the feature enabled/disabled. Thanks for the tip, I'll look into adding such a test in a future PR.

**Pull request merged**